### PR TITLE
rationalise urls, add year path parameter

### PR DESCRIPTION
**Pull request by wavemechanics**
### What

* change "census" to "query"
* get rid of /dev/hello prefix; make metadata, ckmeans, ckmeansratio, and query top level
* require year in ckmeans, ckmeansratio and query
* add ckmeansratio to geodata cli

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -423,16 +441,16 @@ func HandlerWithOptions(si ServerInterface, options ChiServerOptions) http.Handl
...
-		r.Get(options.BaseURL+"/dev/ckmeans", wrapper.GetDevCkmeans)
+		r.Get(options.BaseURL+"/ckmeans/{year}", wrapper.GetCkmeansYear)
 	})
 	r.Group(func(r chi.Router) {
-		r.Get(options.BaseURL+"/dev/ckmeansratio", wrapper.GetDevCkmeansratio)
+		r.Get(options.BaseURL+"/ckmeansratio/{year}", wrapper.GetCkmeansratioYear)
 	})
 	r.Group(func(r chi.Router) {
-		r.Get(options.BaseURL+"/dev/hello/{dataset}", wrapper.GetDevHelloDataset)
+		r.Get(options.BaseURL+"/metadata", wrapper.GetMetadata)
```
Comment (VivianAllen) [inline api/api.go]: OOS for this PR, but @stmuk - reckon a `year` parameter for metadata will be needed eventually?

**Review by VivianAllen — CHANGES_REQUESTED** [review]
cleaner and easier to understand. I think its worth adding changes to the endpoint values in inttests/main.go to match the new ones?

File: api/api.go (hunk 1 above)
Comment (stmuk) [reply api/api.go]: yes

Comment (wavemechanics) [reply api/api.go]: I plan to do that when I've figured out the gorm way of constructing queries.

**Review by VivianAllen — DISMISSED** [review]

**Review by VivianAllen — DISMISSED** [review]
nice!

File: pkg/geodata/ckmeans.go [hunk 2]
```diff
@@ -27,15 +27,15 @@ AND geo.valid
 
 -- the category we are interested in
 AND nomis_category.long_nomis_code = $2
-AND nomis_category.year = 2011
+AND nomis_category.year = $3
```
Comment (VivianAllen) [inline pkg/geodata/ckmeans.go:30]: as @stmuk points out,  would it make more sense to have `nomis_category.year = data_ver.year` and `data_ver.year = $3` here?

File: pkg/geodata/ckmeans.go [hunk 3]
```diff
@@ -128,15 +130,15 @@ AND geo.valid
 
 -- the category we are interested in
 AND nomis_category.long_nomis_code IN ($2, $3)
-AND nomis_category.year = 2011
+AND nomis_category.year = $4
```
Comment (VivianAllen) [inline pkg/geodata/ckmeans.go:132]: as above, should this be `nomis_cateogory.year = data_ver.year` and `data_ver.year = $4`?

**Review by stmuk — APPROVED** [review]

**Pull request merged**